yt_dlp/extractor/berufetv.py

```diff
@@ -0,0 +1,79 @@
+from yt_dlp.extractor.common import InfoExtractor
+from yt_dlp.utils import ExtractorError, mimetype2ext, traverse_obj
+
+
+class BerufeTVIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?web\.arbeitsagentur\.de/berufetv/[a-z\-/]+/film;filmId=(?P<id>[a-zA-Z\d\-_]+)'
+    _TESTS = [{
+        'url': 'https://web.arbeitsagentur.de/berufetv/studienberufe/wirtschaftswissenschaften/wirtschaftswissenschaften-volkswirtschaft/film;filmId=DvKC3DUpMKvUZ_6fEnfg3u',
+        'md5': '041b6432ec8e6838f84a5c30f31cc795',
+        'info_dict': {
+            'id': 'DvKC3DUpMKvUZ_6fEnfg3u',
+            'ext': 'mp4',
+            'title': 'Volkswirtschaftslehre',
+            'description': 'md5:6bd87d0c63163480a6489a37526ee1c1',
+            'categories': ['Studien&shy;beruf'],
+            'tags': ['Studienfilm'],
+            'duration': 602.440,
+            'thumbnail': 'https://asset-out-cdn.video-cdn.net/private/videos/DvKC3DUpMKvUZ_6fEnfg3u/thumbnails/active',
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        movie_metadata = self._download_json(
+            'https://rest.arbeitsagentur.de/infosysbub/berufetv/pc/v1/film-metadata',
+            video_id, 'Downloading JSON metadata',
+            headers={'X-API-Key': '79089773-4892-4386-86e6-e8503669f426'}, fatal=False)
+
+        meta = next(
+            item for item in movie_metadata.get('metadaten') if video_id == item.get('miId')
+        ) if movie_metadata else {}
+
+        video = self._download_json(
+            'https://d.video-cdn.net/play/player/8YRzUk6pTzmBdrsLe9Y88W/video/%s' % (video_id),
+            video_id, 'Downloading video JSON')
+
+        video_sources = traverse_obj(video, ['videoSources', 'html'])
+
+        if not video_sources:
+            raise ExtractorError('Failed to obtain video source list')
+
+        formats = []
+        subtitles = {}
+
+        for key, source in video_sources.items():
```

Comment:
is this error expected?
```suggestion
        video_sources = traverse_obj(video, ['videoSources', 'html'])

        if not video_sources:
            raise ExtractorError('Failed to obtain video source list', expected=True)

        formats = []
        subtitles = {}

        for key, source in video_sources.items():
```
Otherwise, just do
```suggestion
        formats, subtitles = [], {}
        for key, source in video['videoSources']['html'].items():
```